File: features/transaction/transactiondata/src/main/java/com/tokopedia/transactiondata/entity/request/DataCheckoutRequest.java
package com.tokopedia.transactiondata.entity.request;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

/**
 * @author anggaprasetiyo on 05/03/18.
 */

public class DataCheckoutRequest {
    @SerializedName("address_id")
    @Expose
    public int addressId;
    @SerializedName("shop_products")
    @Expose
    public List<ShopProductCheckoutRequest> shopProducts = new ArrayList<>();

    public DataCheckoutRequest() {
    }

    private DataCheckoutRequest(Builder builder) {
        addressId = builder.addressId;
        shopProducts = builder.shopProducts;
    }


    public static final class Builder {
        private int addressId;
        private List<ShopProductCheckoutRequest> shopProducts;

        public Builder() {
        }

        public Builder addressId(int val) {
            addressId = val;
            return this;
        }

        public Builder shopProducts(List<ShopProductCheckoutRequest> val) {
            shopProducts = val;
            return this;
        }

        public DataCheckoutRequest build() {
            return new DataCheckoutRequest(this);
        }
    }
}

File: features/transaction/checkout/src/main/java/com/tokopedia/checkout/data/model/request/checkout/old/DataCheckoutRequest.kt
package com.tokopedia.checkout.data.model.request.checkout.old

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
data class DataCheckoutRequest(
        @SerializedName("address_id")
        var addressId: String? = null,
        @SerializedName("shop_products")
        var shopProducts: List<ShopProductCheckoutRequest>? = ArrayList()
) : Parcelable